box-browse-sdk/src/main/java/com/box/androidsdk/browse/fragments/BoxSearchFragment.java

```diff
@@ -1,43 +1,261 @@
             mLimit = getArguments().getInt(ARG_LIMIT, DEFAULT_LIMIT);
+            mSearchQuery = getArguments().getString(OUT_QUERY, null);
+            mParentFolder = (BoxFolder) getArguments().getSerializable(EXTRA_PARENT_FOLDER);
         }
         if (savedInstanceState != null) {
             mOffset = savedInstanceState.getInt(OUT_OFFSET);
+            mSearchQuery = savedInstanceState.getString(OUT_QUERY, null);
+            mSearchFilters = (BoxSearchFilters) savedInstanceState.getSerializable(EXTRA_SEARCH_FILTERS);
+        }
+
+        if (mSearchFilters == null) {
+            mSearchFilters = new BoxSearchFilters();
+        }
+
+        mItemTypeToExtensions = new HashMap<BoxSearchFilters.ItemType, String[]>();
+        mItemTypeToExtensions.put(BoxSearchFilters.ItemType.Audio, ThumbnailManager.AUDIO_EXTENSIONS_ARRAY);
+        mItemTypeToExtensions.put(BoxSearchFilters.ItemType.BoxNote, ThumbnailManager.BOXNOTE_EXTENSIONS_ARRAY);
+        mItemTypeToExtensions.put(BoxSearchFilters.ItemType.Document, ThumbnailManager.DOCUMENTS_EXTENSIONS_ARRAY);
+        mItemTypeToExtensions.put(BoxSearchFilters.ItemType.Image, ThumbnailManager.IMAGE_EXTENSIONS_ARRAY);
+        mItemTypeToExtensions.put(BoxSearchFilters.ItemType.Pdf, ThumbnailManager.PDF_EXTENSIONS_ARRAY);
+        mItemTypeToExtensions.put(BoxSearchFilters.ItemType.Presentation, ThumbnailManager.PRESENTATION_EXTENSIONS_ARRAY);
+        mItemTypeToExtensions.put(BoxSearchFilters.ItemType.Spreadsheet, ThumbnailManager.SPREADSHEET_EXTENSIONS_ARRAY);
+        mItemTypeToExtensions.put(BoxSearchFilters.ItemType.Video, ThumbnailManager.VIDEO_EXTENSIONS_ARRAY);
+
+        mRequest = null;
+        mRecentSearches = fetchRecentSearches();
+        mRecentSearchesAdapter = new BoxRecentSearchAdapter(getActivity(), mRecentSearches, this);
+    }
+
+    @Override
+    protected int getLayout() {
+        return R.layout.box_browsesdk_fragment_search;
+    }
+
+    @Override
+    public View onCreateView(LayoutInflater inflater, ViewGroup container, Bundle savedInstanceState) {
+        View view = super.onCreateView(inflater, container, savedInstanceState);
+
+        if (mSearchRecentsListView != null) {
+            mSearchRecentsListView.setAdapter(mRecentSearchesAdapter);
+
+            // Show recents only if we are not using an old fragment
+            if (mSearchQuery == null) {
+                mSearchRecentsListView.setVisibility(View.VISIBLE);
+                mSearchRecentsListView.setOnItemClickListener(new AdapterView.OnItemClickListener() {
+                    @Override
+                    public void onItemClick(AdapterView<?> parent, View view, int position, long id) {
+                        String query = mRecentSearches.get(position);
+                        Activity activity = getActivity();
+                        if (activity instanceof BoxBrowseActivity) {
+                            ((BoxBrowseActivity)activity).setSearchQuery(query);
```

Comment:
setSearchQuery sounds like it should be a method on an interface, than an activity.